tests/workflow/test_jobs.py

```diff
@@ -163,3 +163,59 @@ def test_run_with_profile():
         assert (
             f"Profiling [{project.unified_dataset().name}] attributes." == all_ops[1].description
         )
+
+
+@mock_api()
+@pytest.mark.parametrize(
+    "project_name,expected_number_of_output_datasets",
+    [
+        ("minimal_schema_mapping", 1),
+        ("minimal_mastering", 36),
+        ("minimal_golden_records", 7),
+        ("minimal_categorization", 8),
+        ("minimal_phone_enrichment", 1),
+    ],
+)
+def test_get_project_output_datasets(project_name, expected_number_of_output_datasets):
+    reference_postfix_dict = {
+        "SCHEMA_MAPPING_RECOMMENDATIONS": [""],
+        "DEDUP": [
+            "",
+            "_dedup_clusters",
+            "_dedup_clusters_with_data",
+            "_dedup_published_clusters_with_data",
+        ],
+        "GOLDEN_RECORDS": [
+            "_golden_records",
+            "_golden_records_draft",
+            "_golden_records_rule_output",
+        ],
+        "CATEGORIZATION": ["", "_all_classifications", "_classifications_average_confidences"],
+    }
+    client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    test_project = client.projects.by_name(project_name)
+    output_dataset_names = [
+        ds.name for ds in workflow.jobs.get_project_output_datasets(test_project)
+    ]
+    # expected number of output datasets for some project types, e.g., DEDUP
+    # is too large to check all dataset names individually
+    # alternatively, the number of expected output datasets is used for testing
+    # at the time this test was developed TBOX TEST instance ran on v2022.010.2
+    if expected_number_of_output_datasets:
+        assert len(output_dataset_names) == expected_number_of_output_datasets
+    # spot-checking to assert expected datasets are within the returned list
+    if test_project.type != "ENRICHMENT":
+        reference_dataset_name = (
+            test_project.unified_dataset().name
+            if not test_project.type == "GOLDEN_RECORDS"
+            else test_project.name
+        )
+        list_of_expected_output_dataset_names = [
+            f"{reference_dataset_name}{postfix}"
+            for postfix in reference_postfix_dict.get(test_project.type, [])
+        ]
+    else:
+        list_of_expected_output_dataset_names = ["validated_phone_numbers_enriched_dataset"]
```

Comment:
I think it would be easier to follow this with a slightly simpler if-else chain (you can also shorten the name of your variable `list_of_expected_output_dataset_names` without losing clarity):
```
    postfixes = reference_postfix_dict.get(test_project.type, [])
    if test_project.type == "ENRICHMENT":
        expected_dataset_names = ["validated_phone_numbers_enriched_dataset"]
    elif test_project.type == "GOLDEN_RECORDS":
        expected_dataset_names = [f"{test_project.name}{pf}" for pf in postfixes]
    else:
        expected_dataset_names = [f"{test_project.unified_dataset().name}{pf}" for pf in postfixes]
```